index.js

```diff
@@ -43,17 +43,37 @@ function logGraphQLDetails (opts, schema, document, context) {
   const queryOps = readOps(document, 'query', opts)
   const mutationOps = readOps(document, 'mutation', opts)
 
+  // operationName can be provided at the request level, or it can be provided by query/mutation.
+  // If it's provided by both, the request level takes precedence.
+  const operationName = context.reply.request.body.operationName || readOperationName(document, opts)
+  const isCurrentOperation = (op) => op.operationName === operationName
+
+  // Runs on a single operation at a time in a batched query, so we need to pull out
+  // the relevant operation from the batch to be able to log variables for it.
+  const isBatch = Array.isArray(context.reply.request.body)
+  const currentBody = isBatch
+    ? context.reply.request.body.find(isCurrentOperation)
+    : context.reply.request.body
+
   context.reply.request.log[opts.logLevel]({
     graphql: {
       queries: queryOps.length > 0 ? queryOps : undefined,
       mutations: mutationOps.length > 0 ? mutationOps : undefined,
-      operationName: context.reply.request.body.operationName,
+      operationName,
       body: opts.buildBody(context),
-      variables: opts.logVariables === true ? context.reply.request.body.variables || null : undefined
+      variables: opts.logVariables === true ? currentBody?.variables || null : undefined
```

Comment:
if we use the `logBody` parameter - we need to provide to the `buildBody` the currentBody var otherwise it will not work.

We should add a dedicated test case for this as it is not covered right now

Reply:
I've added this, but would love feedback on whether this works as you'd hope.